frontend/packages/console-dynamic-plugin-sdk/src/app/components/utils/headings.tsx

```diff
@@ -0,0 +1,210 @@
+          </SplitItem>
+          {badge && (
+            <SplitItem>{<span className="co-m-pane__heading-badge">{badge}</span>}</SplitItem>
+          )}
+        </Split>
+      )}
+      {showHeading && (
+        <h1
+          className={classNames('co-m-pane__heading', { 'co-m-pane__heading--logo': props.icon })}
+        >
+          {props.icon ? (
+            <props.icon obj={data} />
+          ) : (
+            <div className="co-m-pane__name co-resource-item">
+              {kind && <ResourceIcon kind={kind} className="co-m-resource-icon--lg" />}{' '}
+              <span data-test-id="resource-title" className="co-resource-item__resource-name">
+                {resourceTitle}
+                {data?.metadata?.namespace &&
+                  data?.metadata?.ownerReferences?.length &&
+                  managedByComponent(data)}
+              </span>
+              {resourceStatus && (
+                <ResourceStatus additionalClassNames="hidden-xs">
+                  <Status status={resourceStatus} />
+                </ResourceStatus>
+              )}
+            </div>
+          )}
+          {!breadcrumbsFor && !breadcrumbs && badge && (
+            <span className="co-m-pane__heading-badge">{badge}</span>
+          )}
+          {showActions && (
+            <div className="co-actions" data-test-id="details-actions">
+              {hasButtonActions && (
+                <ActionButtons actionButtons={buttonActions.map((a) => a(kindObj, data))} />
+              )}
+              {hasMenuActions && (
+                <ActionsMenu
+                  actions={
+                    _.isFunction(menuActions)
+                      ? menuActions(kindObj, data, extraResources, customData)
+                      : menuActions.map((a) => a(kindObj, data, extraResources, customData))
+                  }
+                />
+              )}
+              {_.isFunction(customActionMenu) ? customActionMenu(kindObj, data) : customActionMenu}
+            </div>
+          )}
+        </h1>
+      )}
+      {props.children}
+    </div>
+  );
+});
+
+export type KebabOptionsCreator = (
+  kindObj: K8sKind,
+  data: K8sResourceKind,
+  extraResources?: { [prop: string]: K8sResourceKind | K8sResourceKind[] },
+  customData?: any,
+) => KebabOption[];
+
+export type ActionButtonsProps = {
+  actionButtons: any[];
+};
+
+export type BreadCrumbsProps = {
+  breadcrumbs: { name: string; path: string }[];
+};
+
+export type PageHeadingProps = {
+  breadcrumbs?: { name: string; path: string }[];
+  breadcrumbsFor?: (obj: K8sResourceKind) => { name: string; path: string }[];
+  buttonActions?: any[];
+  children?: React.ReactChildren;
+  detail?: boolean;
+  kind?: K8sResourceKindReference;
+  kindObj?: K8sKind;
+  menuActions?: Function[] | KebabOptionsCreator; // FIXME should be "KebabAction[] |" refactor pipeline-actions.tsx, etc.
+  managedByComponent: Function;
```

Comment:
@fhlavac some thoughts:

Is there a reason we chose type `Function` for this new prop? Is it because we need to pass the data down to the child like you have here: `(data) => <ManagedByOperatorLink obj={data} />`? 

I'm a bit worried about using type "function"  because you could theoretically pass in a function that doesn't return what we expect (a React component, JSX element, etc). It also more closely couples the SDK with the consumers of this new prop.

Considering data is just `const data = _.get(obj, 'data');`  and `obj` is just a prop we pass into this class, I wonder if we pass in the pre-rendered component directly as a prop (versus wrapped in a callback function). So you'd have:

`managedByComponent: PropTypes.element,`

and then render it in the common heading via

`{ this.props.managedByComponent }`

and its passed into the common heading via

```jsx
const data = _.get(props.obj, 'data');
const managedByComponent = <ManagedByOperatorLink obj={data} />;
return (
  <PageHeadingCommon
    {...props, managedByComponent}
  />
)
```

We could also perhaps move the conditional that determines if we should render the link out of the heading as well, but we need to be careful not to push too much responsibility back on the consumers and risk duplicated code all over.

This approach would also slightly improve future-proofing if additional information needs to be passed to managedByComponent (the heading class will know nothing about it other than it's a component). 

Apologies for the wall of text!